### PR TITLE
[Merged by Bors] - API phase 3: grpc-gateway proof of concept

### PR DESCRIPTION
**Pull request by lrettig**
## Motivation
See #1764

## Changes
- adds a new GRPC gateway json-http service
- adds related flags and configs (for now, there is only a single flag to enable it, and another flag to set the port; it will be enabled for all corresponding new GRPC services that are also enabled)
- starts/stops it in app setup/teardown

## Test Plan
- tests new commandline flags (turning on the service, setting the port)
- tests enabling the service in combination with enabling new GRPC service (at least one is required)
- tests communication between new JSON service and new GRPC service
- E2E test of one JSON service endpoint

### CODE REVIEW

File: cmd/node/node.go [hunk 1]
```diff
@@ -657,6 +662,43 @@ func (app *SpacemeshApp) startServices() {
 	go app.checkTimeDrifts()
 }
 
+func (app *SpacemeshApp) startAPIServices(postClient api.PostAPI, net api.NetworkAPI) {
+	apiConf := &app.Config.API
+
+	// OLD API SERVICES (deprecated)
+	if apiConf.StartGrpcServer || apiConf.StartJSONServer {
```
Comment (antonlerner) [inline cmd/node/node.go:672]: why keep this code then?

**Review by antonlerner — APPROVED** [review]

File: cmd/node/node.go (hunk 1 above)
Comment (lrettig) [reply cmd/node/node.go:672]: Basically, the plan is to gradually enable the new API through this series of PRs, without removing the old API yet. Then once the new API is totally in place, the old one will be removed as part of the last phase. See the plan in #1764. We can't kill off the old API until the new one is fully working. This is what @noamnelke and I had originally discussed.

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by lrettig** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Pull request closed without merging**